src/portable_python/cpython.py

```diff
@@ -211,14 +241,45 @@ def _finalize(self):
                 runez.write(self.install_folder / info_path, contents)
 
         PPG.config.cleanup_folder(self, "cpython-clean-2nd-pass", "cpython-clean")
+        self._lock_site_packages()
 
         py_inspector = PythonInspector(self.install_folder)
         print(py_inspector.represented())
         problem = py_inspector.full_so_report.get_problem(portable=not is_shared)
         runez.abort_if(problem and self.setup.x_debug != "direct-finalize", "Build failed: %s" % problem)
 
-    def _check_venv(self, copies=False):
-        """Verify that the freshly compiled python can create venvs without issue"""
+    def _lock_site_packages(self):
+        """Mark site-packages as read-only, this will force any '-mpip install' to imply --user"""
+        if PPG.config.get_value("cpython-lock-site-packages"):
+            site_packages = self.prefix_lib_folder / "site-packages"
+            if site_packages.exists():
```

Comment:
I would prefer explicitly checking dryrun mode; this is the sort of pattern where a coding error might result in this silently doing nothing.

Reply:
I'll correct the comment, it's not just dryrun mode, if there is no `site-packages/` folder, then I don't think forcing user to create one so we just mark it as read-only would be useful